src/Control/Exception/Safe.hs

```diff
@@ -247,47 +290,47 @@ catchAnyDeep = catchDeep
 -- <https://github.com/fpco/safe-exceptions#quickstart>
 --
 -- @since 0.1.0.0
-catchAsync :: (C.MonadCatch m, Exception e) => m a -> (e -> m a) -> m a
+catchAsync :: HAS_CALL_STACK => (C.MonadCatch m, Exception e) => m a -> (e -> m a) -> m a
 catchAsync = C.catch
 
 -- | 'catchJust' is like 'catch' but it takes an extra argument which
 -- is an exception predicate, a function which selects which type of
 -- exceptions we're interested in.
 --
 -- @since 0.1.4.0
-catchJust :: (C.MonadCatch m, Exception e) => (e -> Maybe b) -> m a -> (b -> m a) -> m a
-catchJust f a b = a `catch` \e -> maybe (throwM e) b $ f e
+catchJust :: HAS_CALL_STACK => (C.MonadCatch m, Exception e) => (e -> Maybe b) -> m a -> (b -> m a) -> m a
+catchJust f a b = withFrozenCallStack a `catch` \e -> maybe (throwM e) b $ f e
```

Comment:
This freezes the `CallStack` for `a`, which is probably not what you want - I'd guess you want to freeze the callstack for `catch`, which would require:

```suggestion
catchJust f a b = withFrozenCallStack catch a (\e -> maybe (throwM e) b $ f e)
```